[DEFECT] ClassCastException when opening contentlet with Block Editor field

### Problem Statement

When attempting to open a contentlet that contains a Block Editor field, the content fails to load for editing. The issue only appears with specific customer data. According to the logs, the data attribute inside a Block Editor structure was saved as a String instead of a Map, causing a ClassCastException during the StoryBlock refresh process.
This prevents editors from modifying the affected contentlets and disrupts regular content operations.

### Steps to Reproduce

1.	Log into a dotCMS instance running 25.11.07 with the affected customer content.
2.	Navigate to Content Search.
3.	Locate a contentlet containing a Block Editor field. (id = 86dc6e4ad903f4e499bd074b92bbdf59)
4.	Attempt to open the content for editing.
5.	Observe that the UI fails to load and check the backend logs.
6.	Note the error:
```
Caused by: java.lang.ClassCastException: class java.lang.String cannot be cast to class java.util.Map 
(java.lang.String and java.util.Map are in module java.base of loader 'bootstrap')
    at com.dotcms.contenttype.business.StoryBlockAPIImpl.refreshStoryBlockMap(StoryBlockAPIImpl.java:229)
    at com.dotcms.contenttype.business.StoryBlockAPIImpl.isRefreshed(StoryBlockAPIImpl.java:207)
    at com.dotcms.contenttype.business.StoryBlockAPIImpl.refreshStoryBlockValueReferences(StoryBlockAPIImpl.java:183)
```

### Acceptance Criteria

•	Contentlets containing Block Editor fields load without throwing ClassCastException.
•	Block Editor data attribute is properly interpreted as a Map, even if previously stored as a String.
•	A migration, fallback, or defensive handling is implemented to avoid crashes when malformed Block Editor data exists.
•	Backend logs show no ClassCastExceptions related to StoryBlockAPIImpl.refreshStoryBlockMap.

### dotCMS Version

25.11.07

### Severity

High - Major functionality broken

--- a/dotCMS/src/main/java/com/dotcms/contenttype/business/StoryBlockAPIImpl.java
+++ b/dotCMS/src/main/java/com/dotcms/contenttype/business/StoryBlockAPIImpl.java
@@ -226,8 +226,29 @@ public class StoryBlockAPIImpl implements StoryBlockAPI {
         boolean refreshed  = false;
         final Map<String, Object> attrsMap = (Map<String, Object>) contentMap.get(ATTRS_KEY);
         if (UtilMethods.isSet(attrsMap)) {
-            final Map<String, Object> dataMap = (Map<String, Object>) attrsMap.get(DATA_KEY);
-            if (UtilMethods.isSet(dataMap)) {
+            final Object dataValue = attrsMap.get(DATA_KEY);
+            if (UtilMethods.isSet(dataValue)) {
+                Map<String, Object> dataMap;
+
+                // Handle case where data is a JSON string instead of a Map
+                if (dataValue instanceof String) {
+                    try {
+                        dataMap = this.toMap(dataValue);
+                    } catch (JsonProcessingException e) {
+                        Logger.warnAndDebug(this.getClass(), String.format(
+                            "Failed to parse data field as JSON for parent contentlet '%s': %s",
+                            parentContentletIdentifier, ExceptionUtil.getErrorMessage(e)), e);
+                        return false;
+                    }
+                } else if (dataValue instanceof Map) {
+                    dataMap = (Map<String, Object>) dataValue;
+                } else {
+                    Logger.warn(this.getClass(), String.format(
+                        "Unexpected data type '%s' for data field in parent contentlet '%s'",
+                        dataValue.getClass().getName(), parentContentletIdentifier));
+                    return false;
+                }
+
                 final String identifier = (String) dataMap.get(IDENTIFIER_KEY);
                 final long languageId = ConversionUtils.toLong(dataMap.get(LANGUAGE_ID_KEY), ()-> APILocator.getLanguageAPI().getDefaultLanguage().getId());
                 if (UtilMethods.isSet(identifier)) {

--- a/dotcms-integration/src/test/java/com/dotcms/contenttype/business/StoryBlockAPITest.java
+++ b/dotcms-integration/src/test/java/com/dotcms/contenttype/business/StoryBlockAPITest.java
@@ -1122,4 +1122,129 @@ public class StoryBlockAPITest extends IntegrationTestBase {
             }
         }
     }
+
+    /**
+     * Method to test: {@link StoryBlockAPI#refreshStoryBlockValueReferences(Object, String)}
+     * Given Scenario: Test the fix for issue-33900 where the data attribute could be a JSON String
+     * instead of a Map, causing a ClassCastException. This test covers three scenarios:
+     * 1. Data as a JSON String (the bug case) - should parse successfully
+     * 2. Data as a Map (normal case) - should process successfully
+     * 3. Data as an invalid type (Integer) - should handle gracefully
+     * 4. Data as an invalid JSON String - should handle gracefully
+     * ExpectedResult: All scenarios should be handled without throwing ClassCastException
+     */
+    @Test
+    public void test_refreshStoryBlockValueReferences_data_field_as_string_vs_map() throws DotDataException, DotSecurityException {
+        final ContentType contentTypeRichText = APILocator.getContentTypeAPI(APILocator.systemUser()).find("webPageContent");
+        final Contentlet richTextContentlet = new ContentletDataGen(contentTypeRichText)
+                .setProperty("title", "Test Content")
+                .setProperty("body", TestDataUtils.BLOCK_EDITOR_DUMMY_CONTENT)
+                .nextPersistedAndPublish();
+
+        final HttpServletRequest oldThreadRequest = HttpServletRequestThreadLocal.INSTANCE.getRequest();
+        final HttpServletResponse oldThreadResponse = HttpServletResponseThreadLocal.INSTANCE.getResponse();
+
+        try {
+            final HttpServletRequest request = new MockAttributeRequest(mock(HttpServletRequest.class));
+            HttpServletRequestThreadLocal.INSTANCE.setRequest(request);
+
+            final HttpServletResponse response = mock(HttpServletResponse.class);
+            HttpServletResponseThreadLocal.INSTANCE.setResponse(response);
+
+            // Test Case 1: Data field as JSON String (the bug case that caused ClassCastException)
+            final String storyBlockJsonWithDataAsString = String.format(
+                    "{" +
+                    "  \"type\": \"doc\"," +
+                    "  \"content\": [" +
+                    "    {" +
+                    "      \"type\": \"dotContent\"," +
+                    "      \"attrs\": {" +
+                    "        \"data\": \"{\\\"identifier\\\":\\\"%s\\\",\\\"languageId\\\":%s}\"" +
+                    "      }" +
+                    "    }" +
+                    "  ]" +
+                    "}",
+                    richTextContentlet.getIdentifier(),
+                    richTextContentlet.getLanguageId()
+            );
+
+            final StoryBlockReferenceResult resultWithString = APILocator.getStoryBlockAPI()
+                    .refreshStoryBlockValueReferences(storyBlockJsonWithDataAsString, "parent-123");
+
+            assertNotNull("Result should not be null for data as String", resultWithString);
+            assertTrue("Should successfully process data field as JSON String", resultWithString.isRefreshed());
+
+            // Test Case 2: Data field as Map (normal/expected case)
+            final String storyBlockJsonWithDataAsMap = String.format(
+                    "{" +
+                    "  \"type\": \"doc\"," +
+                    "  \"content\": [" +
+                    "    {" +
+                    "      \"type\": \"dotContent\"," +
+                    "      \"attrs\": {" +
+                    "        \"data\": {" +
+                    "          \"identifier\": \"%s\"," +
+                    "          \"languageId\": %s" +
+                    "        }" +
+                    "      }" +
+                    "    }" +
+                    "  ]" +
+                    "}",
+                    richTextContentlet.getIdentifier(),
+                    richTextContentlet.getLanguageId()
+            );
+
+            final StoryBlockReferenceResult resultWithMap = APILocator.getStoryBlockAPI()
+                    .refreshStoryBlockValueReferences(storyBlockJsonWithDataAsMap, "parent-456");
+
+            assertNotNull("Result should not be null for data as Map", resultWithMap);
+            assertTrue("Should successfully process data field as Map", resultWithMap.isRefreshed());
+
+            // Test Case 3: Data field as invalid type (Integer) - should handle gracefully
+            final String storyBlockJsonWithDataAsInteger =
+                    "{" +
+                    "  \"type\": \"doc\"," +
+                    "  \"content\": [" +
+                    "    {" +
+                    "      \"type\": \"dotContent\"," +
+                    "      \"attrs\": {" +
+                    "        \"data\": 12345" +
+                    "      }" +
+                    "    }" +
+                    "  ]" +
+                    "}";
+
+            final StoryBlockReferenceResult resultWithInteger = APILocator.getStoryBlockAPI()
+                    .refreshStoryBlockValueReferences(storyBlockJsonWithDataAsInteger, "parent-789");
+
+            assertNotNull("Result should not be null for data as Integer", resultWithInteger);
+            // Should return false in isRefreshed since data type is unexpected
+            assertFalse("Should handle unexpected data type gracefully", resultWithInteger.isRefreshed());
+
+            // Test Case 4: Data field as invalid JSON String - should handle gracefully
+            final String storyBlockJsonWithInvalidJsonString =
+                    "{" +
+                    "  \"type\": \"doc\"," +
+                    "  \"content\": [" +
+                    "    {" +
+                    "      \"type\": \"dotContent\"," +
+                    "      \"attrs\": {" +
+                    "        \"data\": \"this is not valid json {{{\"" +
+                    "      }" +
+                    "    }" +
+                    "  ]" +
+                    "}";
+
+            final StoryBlockReferenceResult resultWithInvalidJson = APILocator.getStoryBlockAPI()
+                    .refreshStoryBlockValueReferences(storyBlockJsonWithInvalidJsonString, "parent-999");
+
+            assertNotNull("Result should not be null for invalid JSON string", resultWithInvalidJson);
+            // Should return false since JSON parsing failed
+            assertFalse("Should handle invalid JSON string gracefully", resultWithInvalidJson.isRefreshed());
+
+        } finally {
+            HttpServletRequestThreadLocal.INSTANCE.setRequest(oldThreadRequest);
+            HttpServletResponseThreadLocal.INSTANCE.setResponse(oldThreadResponse);
+        }
+    }
 }

